File: ruby/testcode/ssti_037.rb
require_relative 'shared'

begin
  require 'tilt'
rescue LoadError
end

# vuln-code-snippet start ruby_ssti_tilt_file_only
def handler(req)
  output = Tilt.new('views/fixed.html.erb').render(Object.new, { name: req.param('name') }) # vuln-code-snippet safe-line ruby_ssti_tilt_file_only
  BenchmarkResponse.html(output)
end
# vuln-code-snippet end ruby_ssti_tilt_file_only

File: php/testcode/ssti_037.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssti_smarty_compiled_file
function ssti037(BenchmarkRequest $req): BenchmarkResponse {
    $smarty = new Smarty();
    $smarty->display('compiled/user_card.tpl'); // vuln-code-snippet safe-line php_ssti_smarty_compiled_file
    return BenchmarkResponse::ok('rendered');
}
// vuln-code-snippet end php_ssti_smarty_compiled_file
